doc/api/errors.md

```diff
@@ -2266,6 +2266,13 @@ The V8 platform used by this instance of Node.js does not support creating
 Workers. This is caused by lack of embedder support for Workers. In particular,
 this error will not occur with standard builds of Node.js.
 
+<a id="ERR_MODULE_LINK_MISMATCH"></a>
+
+### `ERR_MODULE_LINK_MISMATCH`
+
+A module can not be linked due to a mismatch of the requested modules, and the
+list of given dependency modules.
```

Comment:
It might even make sense to split this into two separate errors:

1. ERR_MODULE_PARTIAL_LINK
2. ERR_MODULE_INVALID_LINK

Where the invalid case is for more than expected link arguments, but the case of fewer than expected is actually a different thing - it's missing linkage so a partial link, which I hope in future can be supported when enabled.